### PR TITLE
New term fix

### PR DESCRIPTION
**Pull request by j-jasik**
This PR:
- Adds student role for anybody who signs into an open session as a student
- Pulls in student courses when they are activated
- Pulls in courses and profs from banner on new term startup (don't have to wait for cron to run)
- A couple of small refactors.

### CODE REVIEW

File: sciencelabs/db_repository/user_functions.py [hunk 1]
```diff
@@ -79,12 +79,17 @@ def get_unique_sessions_attended(self, student_id, semester_id):
...
+        student_courses = db_session.query(Course_Table)\
             .filter(student_id == user_course_Table.user_id)\
             .filter(user_course_Table.course_id == Course_Table.id)\
             .filter(Course_Table.semester_id == Semester_Table.id)\
             .filter(Semester_Table.id == semester_id)\
             .all()
+        student_courses_to_return = []
+        for course in student_courses:
+            student_courses_to_return.append(course)
+        return student_courses_to_return
```
Comment (c-schwarze) [inline sciencelabs/db_repository/user_functions.py]: whats the point of 88-91? Is `student_courses` not already an array?

File: sciencelabs/db_repository/user_functions.py [hunk 2]
```diff
@@ -169,8 +174,31 @@ def activate_existing_user(self, username):
             .filter(User_Table.username == username)\
             .one()
         user.deletedAt = None
+        user_roles = self.get_user_roles(user.id)
+        for role in user_roles:
+            if role.name == 'Student':
```
Comment (c-schwarze) [inline sciencelabs/db_repository/user_functions.py]: I almost feel like 177-179 should be a method? `is_user_student()` or something? Or isn't there a method already that checks roles?

File: sciencelabs/db_repository/user_functions.py [hunk 3]
```diff
@@ -169,8 +174,31 @@ def activate_existing_user(self, username):
...
+                self.set_student_courses(user)
         db_session.commit()
 
+    def set_student_courses(self, student):
+        active_semester = self.get_active_semester()
+        student_lab_courses = self.get_student_courses(student.id, active_semester.id)
+        if student_lab_courses == []:  # No courses in db so check banner
+            student_courses = self.wsapi.get_student_courses(student.username)
+            for key, course in student_courses.items():
+                if db_session.query(CourseCode_Table).filter(CourseCode_Table.courseNum == course['cNumber'])\
```
Comment (c-schwarze) [inline sciencelabs/db_repository/user_functions.py]: Line 189 probably needs a comment

File: sciencelabs/db_repository/user_functions.py (hunk 2 above)
Comment (c-schwarze) [reply sciencelabs/db_repository/user_functions.py]: I was thinking `self.slc.check_roles_and_route(['Administrator'])`, but that probably can't be used here.

File: sciencelabs/db_repository/user_functions.py (hunk 3 above)
Comment (j-jasik) [reply sciencelabs/db_repository/user_functions.py]: commented

File: sciencelabs/db_repository/user_functions.py (hunk 1 above)
Comment (j-jasik) [reply sciencelabs/db_repository/user_functions.py]: ... I don't know. Fixed

File: sciencelabs/db_repository/user_functions.py (hunk 2 above)
Comment (j-jasik) [reply sciencelabs/db_repository/user_functions.py]: I pulled it out to a method.

**Pull request merged**